### PR TITLE
Use search functionality for collection listing, archive, etc [WIP]

### PR DESCRIPTION
**Pull request by tlrobinson**
Frontend TODO:

- [ ] make sure colors/icons are correct
- [ ] fix `CollectionItemsLoader` `entityQuery` (`s/collectionId/collection_id/`)

Frontend TODO stretch:

- [ ] implement automatic entity list reloading
  - [ ] invalidate request states for `entityQuery`s where an action could affect the content
    - [ ] need to look at old and new state
  - [ ] have `EntityListLoader` automatically reload a list if it's request state becomes invalid

Backend TODO:

- [x] change `type: "card"` to `type: "question"`
- [x] make `q` optional
- [ ] make `collection_id=root` work
- [ ] segments/metrics:
  - either add to collections, or don't show if `collection_id` is provided
- [ ] return `collection_position` (and `archived`, `collection`, etc?) in listed objects

### CODE REVIEW

**Comment by tlrobinson** [comment]
@senior I made `q` optional so we can the "search" endpoint for collections listing and archive without providing a search string.

Also, before merging this we need to sort out `metrics` and `segments`. I think there are two options:

1. go ahead and add segments/metrics to collections
2. don't include segments/metrics in the results if a `collection_id` is provided

**Comment by senior** [comment]
@tlrobinson We already don't include segments/metrics when `collection_id` is provided, let me know if that's not working for you.

I get get collection_position, archived and collection included in the response too.

**Comment by tlrobinson** [comment]
@senior ok thanks. I think I thought it wasn't working because i was using `collectionId`.

In case you missed it we also need `collection_id=root` to work

File: src/metabase/api/search.clj [hunk 1]
```diff
@@ -42,7 +42,9 @@
 (s/defn ^:private merge-name-search
   "Add case-insensitive name query criteria to `query-map`"
   [query-map {:keys [search-string]} :- SearchContext]
-  (h/merge-where query-map [:like :%lower.name search-string]))
+  (if (empty? search-string)
```
Comment (camsaul) [inline src/metabase/api/search.clj]: https://github.com/bbatsov/clojure-style-guide#nil-punning

**Pull request closed without merging**